gobblin-modules/gobblin-compliance/src/main/java/gobblin/compliance/DatasetDescriptorImpl.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.compliance;
+
+import java.util.List;
+
+import com.google.common.base.Splitter;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+
+/**
+ * This is the default implementation of {@link DatasetDescriptor}
+ *
+ * @author adsharma
+ */
+public class DatasetDescriptorImpl implements DatasetDescriptor {
+  private static final Splitter DOT_SPLITTER = Splitter.on(".").omitEmptyStrings().trimResults();
+  private String complianceId;
+
+  /**
+   *
+   * @param descriptor is the JsonString corresponding to the value of dataset.descriptor.
+   * @param identifierField is the path to the identifier in the dataset.descriptor.
+   */
+  public DatasetDescriptorImpl(String descriptor, String identifierField) {
+    JsonObject descriptorObject = new JsonParser().parse(descriptor).getAsJsonObject();
+    List<String> list = DOT_SPLITTER.splitToList(identifierField);
+    for (int i = 0; i < list.size() - 1; i++) {
+      descriptorObject = descriptorObject.getAsJsonObject(list.get(i));
```

Comment:
It's good to have some defensive programming if the path does not exist (i.e. descriptorObject == null). What is the expected behavior  -- throw an exception or ignore the dataset?